kernelci/cli/user.py

```diff
@@ -55,57 +57,50 @@ def find(attributes, config, api, indent):
 @click.argument('username')
 @Args.config
 @Args.api
-@Args.indent
-def token(username, config, api, indent):
+@catch_http_error
+def token(username, config, api):
     """Create a new API token using a user name and password"""
-    password = getpass.getpass()
     configs = kernelci.config.load(config)
     api_config = configs['api'][api]
     api = kernelci.api.get_api(api_config)
-    user_token = api.create_token(username, password)
-    click.echo(json.dumps(user_token, indent=indent))
-
-
-@kci_user.group(name='password')
-def user_password():
-    """Manage user passwords"""
+    password = getpass.getpass()
+    api_token = api.create_token(username, password)
+    click.echo(api_token['access_token'])
 
 
 @kci_user.command(secrets=True)
-@click.option('--username')
-@click.option('--email')
-@click.option('--group', multiple=True, help="User group(s)")
+@click.argument('attributes', nargs=-1)
+@click.option('--username', help="Username of the user to update (admin only)")
 @Args.config
 @Args.api
-@Args.indent
-def update(username, email, config,  # pylint: disable=too-many-arguments
-           api, secrets, group, indent):
-    """Update own user account"""
-    user = {}
-    if username:
-        user['username'] = username
-    if email:
-        user['email'] = email
-    if group:
-        user['groups'] = group
-    if not user:
-        raise click.ClickException("Sorry, nothing to update")
+@catch_http_error
+def update(attributes, username, config, api, secrets):
+    """Update user account data with the provided attributes"""
+    fields = split_attributes(attributes)
```

Comment:
```
$ ./kci user update
Traceback (most recent call last):
  File "/home/jeny/kernelCI/kernelci-core_env/src/gctucker/kernelci-core/./kci", line 28, in <module>
    kci()  # pylint: disable=no-value-for-parameter
  File "/home/jeny/kernelCI/kernelci-core_env/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/jeny/kernelCI/kernelci-core_env/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/jeny/kernelCI/kernelci-core_env/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jeny/kernelCI/kernelci-core_env/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jeny/kernelCI/kernelci-core_env/src/gctucker/kernelci-core/kernelci/cli/__init__.py", line 132, in invoke
    super().invoke(ctx)
  File "/home/jeny/kernelCI/kernelci-core_env/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jeny/kernelCI/kernelci-core_env/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/jeny/kernelCI/kernelci-core_env/src/gctucker/kernelci-core/kernelci/cli/__init__.py", line 60, in call
    return func(*args, **kwargs)
  File "/home/jeny/kernelCI/kernelci-core_env/src/gctucker/kernelci-core/kernelci/cli/user.py", line 79, in update
    fields = split_attributes(attributes)
  File "/home/jeny/kernelCI/kernelci-core_env/src/gctucker/kernelci-core/kernelci/cli/__init__.py", line 197, in split_attributes
    for attribute in attributes:
TypeError: 'NoneType' object is not iterable
```

A check seems to be missing here.
```
if attributes:
    fields = split_attributes(attributes)
```

Reply:
Yes there's another PR to fix this: #2185

Ah it's merged already, so I'll rebase this PR.